packages/auth/src/AuthProvider.tsx

```diff
@@ -113,52 +140,67 @@ type AuthProviderState = {
  *  </AuthProvider>
  * ```
  */
-export class AuthProvider extends React.Component<
-  AuthProviderProps,
-  AuthProviderState
-> {
-  static defaultProps = {
-    skipFetchCurrentUser: false,
-  }
-
-  state: AuthProviderState = {
-    loading: true,
-    isAuthenticated: false,
-    userMetadata: null,
-    currentUser: null,
-    hasError: false,
-  }
-
-  rwClient: AuthClient
-
-  constructor(props: AuthProviderProps) {
-    super(props)
-    this.rwClient = createAuthClient(
+export const AuthProvider: React.FC<PropsWithChildren<AuthProviderProps>> = (
+  props: PropsWithChildren<AuthProviderProps>
+) => {
+  const skipFetchCurrentUser = props.skipFetchCurrentUser || false
+
+  const [hasRestoredState, setHasRestoredState] = useState(false)
+
+  const [authProviderState, setAuthProviderState] = useState(
+    defaultAuthProviderState
+  )
+
+  const [rwClient, setRwClient] = useState<AuthClient | null>(null)
```

Comment:
```suggestion
  const [rwClient, setRwClient] = useState<AuthClient>()
```

Just simplifying the code a bit. If I'm reading things correctly you're not actually using the `null` value anywhere.

Reply:
Yup, just a stylistic thing of explicit `null` rather than implicit `undefined` but no big. Changed.